Fixes the issue with a date change firing mulitple change events

darylldoyle opened the pull request:
This PR fixes the issue with a date change firing multiple change events as reported in #1233 

Fix is pulled from [this issue](https://github.com/vizjerai/bootstrap-datepicker/issues/4) and [this commit](https://github.com/vizjerai/bootstrap-datepicker/commit/db20cae1d6ada09c9693fb780567388508b3ff57)

I've tested this locally and all seems to be working OK with it.

hebbet commented:
thanks. could you also fix the test?

acrobat commented:
When the PR is ready we should merge this in the 1.4 branch, so we can push this fix the next bugfix release

darylldoyle commented:
No worries, will update it tonight when I'm home!

acrobat commented on tests/suites/events.js:
```diff
@@ -301,7 +301,7 @@ test('setDate: triggers change and changeDate events', function(){
     this.dp.setDate(new Date(2011, 2, 5));
 
-    equal(triggered_change, 2);
+    equal(triggered_change, 0);
```
Should the change event not be called once? Or do we just want to leave out the change event and only have a changeDate event, but that would mean a BC break

darylldoyle replied:
The reason for taking it out completely is that the change event will be fired on the base element anyway won't it? Therefore if someone is targeting the base element with an `on.('change', ...)` handler, it will always be fired more than once. 

On the other hand, if someone want to specifically listen for a date change, they can listen for the `changeDate` event.

Would it be worth moving this PR to target a 2.0 branch if using semver or otherwise 1.4 to account for the BC break?

acrobat replied:
Looking at the result of the test, the change event isn't fired anymore on the input field, if we just fire change once it wouldn't be a BC break. The issue is that it fires the change event more then once, if it would just fire once this bug is fixed without the BC break

darylldoyle commented:
I think this may solve it. I did a bit of profiling and it seems that the `update` method is called pretty much anytime something is done through the public API. Therefore I've removed the change event from the `clearDates` method and added it to the bottom of the `update` method.

You may have more insight on whether this will or wont work though!

vsn4ik commented:
:+1: for merge.

darylldoyle commented:
Any news on this one?

acrobat commented:
Looks good @darylldoyle! Thanks!
